lib/providers/device-log-provider.ts

```diff
@@ -0,0 +1,39 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+export class DeviceLogProvider implements Mobile.IDeviceLogProvider {
+	private static LINE_REGEX = /.\/(.+?)\s*\(\s*(\d+?)\): (.*)/;
+
+	constructor(private $devicePlatformsConstants: Mobile.IDevicePlatformsConstants,
+				private $logger: ILogger) { }
+
+	public logData(lineText: string, platform: string, deviceIdentifier: string): void {
+		if(!platform || platform === this.$devicePlatformsConstants.iOS) {
+			this.$logger.out(lineText);
+		} else if (platform === this.$devicePlatformsConstants.Android) {
+			let log = this.getConsoleLogFromLine(lineText);
+			if(log) {
+				if(log.tag) {
+					this.$logger.out("%s: %s", log.tag, log.message);
+				} else {
+					this.$logger.out(log.message);
+				}
+			}
+		}
+	}
+
+	private getConsoleLogFromLine(lineText: String): any {
+		let acceptedTags = ["chromium", "Web Console", "JS"];
+		//sample line is "I/Web Console(    4438): Received Event: deviceready at file:///storage/emulated/0/Icenium/com.telerik.TestApp/js/index.js:48"
```

Comment:
I think this comment would be way more useful if moved near the LINE_REGEX definition.